### PR TITLE
apps: Split eligble clusters logic out from volume entry create

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Changes split out cluster filtering logic, clean up and compact the filtering logic, and add check fo bad clusters.

### CODE REVIEW

File: apps/glusterfs/volume_entry.go [hunk 1]
```diff
@@ -615,6 +615,12 @@ func eligibleClusters(db *bolt.DB, v *VolumeEntry,
 			switch {
 			case v.Info.Block && c.Info.Block:
 			case !v.Info.Block && c.Info.File:
+			case !(c.Info.Block || c.Info.File):
+				// bad cluster config!
+				err = fmt.Errorf("Cluster %v lacks both block and file flags",
+					clusterId)
+				logger.LogError(err.Error())
+				return err
```
Comment (obnoxxx) [inline apps/glusterfs/volume_entry.go]: I don't think we want this: The administrator might deliberately have mark this cluster to not create any volumes (maybe temporarily). So I suggest to drop this commit.

**Comment by phlogistonjohn** [comment]
> I don't want us to treat the condition of c.Info.File == false && c.Info.Block == False as an error and bail out on it: It is a valid configuration that the admin might have deliberately chosen to disable volumes to be created on that cluster (maybe temporarily). So I propose to drop this last commit.

Do you mind if I just switch it to a info log line? I think it is worth having that extra knowledge about why the cluster was skipped. This issue bit me hard when I was working on the pr to update doc/admin/install-kubernetes.md

**Comment by obnoxxx** [comment]
> > I don't want us to treat the condition of c.Info.File == false && c.Info.Block == False as an error and bail out on it: It is a valid configuration that the admin might have deliberately chosen to disable volumes to be created on that cluster (maybe temporarily). So I propose to drop this last commit.
> 
> Do you mind if I just switch it to a info log line? I think it is worth having that extra knowledge about why the cluster was skipped. This issue bit me hard when I was working on the pr to update doc/admin/install-kubernetes.md

Sure, that's a good idea.

File: apps/glusterfs/volume_entry.go [hunk 2]
```diff
@@ -615,6 +615,10 @@ func eligibleClusters(db *bolt.DB, v *VolumeEntry,
 			switch {
 			case v.Info.Block && c.Info.Block:
 			case !v.Info.Block && c.Info.File:
+			case !(c.Info.Block || c.Info.File):
+				// possibly bad cluster config
+				logger.Info("Cluster %v lacks both block and file flags",
+					clusterId)
```
Comment (obnoxxx) [inline apps/glusterfs/volume_entry.go]: I think you should return nil here, right?

**Comment by obnoxxx** [comment]
retest this please

**Pull request merged**